zio-http-cli/src/main/scala/zio/http/endpoint/cli/CliRequest.scala

```diff
@@ -2,27 +2,39 @@ package zio.http.endpoint.cli
 
 import zio._
 import zio.json.ast._
+import java.nio.file.Path
 
 import zio.http._
+import zio.cli._
+import java.nio.channels.FileChannel
+
+import zio.stream.{ZSink, ZStream}
+
+import java.io.File
+import scala.io.Source
+import java.io.IOException
+
+
+/**
+ * Represents a Request. The body parameter allows implementation of multipart form data and the retrieval of a body
+ * from a file or an URL.
+ */
 
 private[cli] final case class CliRequest(
-  url: URL,
-  method: Method,
+  body: Chunk[Either[Either[(String, Path, MediaType), String], FormField]],
```

Comment:
At the point of `Either[Either[...]]`, it's worth introducing a `sealed trait` to ease code comprehension and maintenance. Also, it seems you moved from `Json` to `String` for the body. I don't see how that can work if the body must be generated incrementally by each option / argument.

Reply:
Certainly, I would call it BodyConstructor. Are you referring to this? https://github.com/zio/zio-http/blob/76300ea8e16cb4fc603f3d0ec746028951565714/zio-http-cli/src/main/scala/zio/http/endpoint/cli/HttpOptions.scala#L59
Should it be instead of `textField` `binaryField`? `a` would be `Json` but it specifies only a FormField not the whole body. The `BodyConstructor[(String, Path, MediaType), String, FormField]` would represent  `BodyConstructor[from File, from Url, FormField from Json]` that would be converted to a FormField depending on the specification and that `Chunk[FormField]` would be used to specify the (possibly multipart) body.